plugins/discovery-ec2/build.gradle

```diff
@@ -56,12 +58,34 @@ task writeTestJavaPolicy {
       throw new GradleException("failed to create temporary directory [${tmp}]")
     }
     final File javaPolicy = file("${tmp}/java.policy")
-    javaPolicy.write(
-      [
-        "grant {",
-        "  permission java.util.PropertyPermission \"com.amazonaws.sdk.ec2MetadataServiceEndpointOverride\", \"write\";",
-        "};"
-      ].join("\n"))
+    if (BuildParams.inFipsJvm) {
+      javaPolicy.write(
```

Comment:
There might be a way to read the policy as we set it in BuildPlugin and just write that to the file adding the 1 necessary grant but I'm not aware how we could do this, so I went with the easy way.